From 71feadca0736882ca1f7d52c26d2c988f0c5afed Mon Sep 17 00:00:00 2001
From: rgret-dev <github@rgret.dev>
Date: Tue, 28 Apr 2026 18:01:38 -0500
Subject: [PATCH 01/10] UI/Gtk: Support keypad buttons for zoom actions

The Qt UI supports this so GTK should have it, too :^)

Backported from upstream commit 19e74815ed78.
---
 UI/Gtk/BrowserWindow.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/UI/Gtk/BrowserWindow.cpp b/UI/Gtk/BrowserWindow.cpp
index 5f89ed71..8434ab64 100644
--- a/UI/Gtk/BrowserWindow.cpp
+++ b/UI/Gtk/BrowserWindow.cpp
@@ -259,9 +259,9 @@ void BrowserWindow::setup_keyboard_shortcuts()
     set_accels("win.find-close", { "Escape" });
     set_accels("win.go-back", { "<Alt>Left" });
     set_accels("win.go-forward", { "<Alt>Right" });
-    set_accels("win.zoom-in", { "<Ctrl>equal", "<Ctrl>plus" });
-    set_accels("win.zoom-out", { "<Ctrl>minus" });
-    set_accels("win.zoom-reset", { "<Ctrl>0" });
+    set_accels("win.zoom-in", { "<Ctrl>equal", "<Ctrl>plus", "<Ctrl>KP_Add" });
+    set_accels("win.zoom-out", { "<Ctrl>minus", "<Ctrl>KP_Subtract" });
+    set_accels("win.zoom-reset", { "<Ctrl>0", "<Ctrl>KP_0" });
     set_accels("win.fullscreen", { "F11" });
     set_accels("win.quit", { "<Ctrl>q" });
     set_accels("win.new-window", { "<Ctrl>n" });
